### PR TITLE
0x6e/ios keymanagement

### PR DESCRIPTION
**Pull request by commjoen**
First iteration for keymanagement in iOS (https://github.com/OWASP/owasp-mstg/issues/354)

### CODE REVIEW

File: Document/0x06d-Testing-Data-Storage.md [hunk 1]
```diff
@@ -39,27 +39,42 @@ Since iOS 7, the default data protection class is "Protected Until First User Au
...
+- `kSecAccessControlDevicePasscode`: only access the item using a passcode
+- `kSecAccessControlTouchIDAny` : access the item using one of your fingerprints registered to TouchID. Adding or removing a fingerprint will not invalidate the item.
+- `kSecAccessControlTouchIDCurrentSet`: access the item using one of your fingerprints registered to TouchID. Adding or removing a fingerprint _will_ invalidate the item.
+- `kSecAccessControlUserPresence`: access the item using either one of the registered fingerprint (using TouchID) or fallback to the PassCode.
+
+Please note that keys secured by TouchID (using `kSecAccessControlTouchIDCurrentSet` or `kSecAccessControlTouchIDAny`) are protected by the Secure Enclave: the keychain only holds a token, but not the actual key. The key resides in the Secure Enclave.
+
+Next, from iOS 9 onward, you can do ECC based signign operations in the Secure Enclave. In that case the private key as well as the cryptographic operations reside within the Secure Enlave. See the remedation chapter for more info on creating the ECC keys.
```
Comment (bolot) [inline Document/0x06d-Testing-Data-Storage.md]: Typo: "signing" (ng ending).
Typo: "Enclave" (missing c).
Might want to rephrase the sentence to say "the cryptographic operations are performed within the Secure Enclave".

File: Document/0x06d-Testing-Data-Storage.md [hunk 2]
```diff
@@ -150,6 +165,41 @@ The following example shows how to create a securely encrypted file using the `c
 
 A generic example for using the KeyChain to store, update or delete data can be found in the official Apple documentation<sup>[12]</sup>.
 
+A sample for using TouchID and passcode protected keys can be found in the official Apple documentaiton <sup><[13]</sup>.
```
Comment (bolot) [inline Document/0x06d-Testing-Data-Storage.md]: Typo: "documentation" (i and t got switched).
Extra `<` before `[13]`.

File: Document/0x06d-Testing-Data-Storage.md [hunk 3]
```diff
@@ -39,27 +39,42 @@ Since iOS 7, the default data protection class is "Protected Until First User Au
...
+By default, each app can only access the Keychain created by itself. Access can however be shared between apps signed by the same developer by using the "access groups" feature (`kSecAttrAccessGroup` see <sup>[https://developer.apple.com/documentation/security/ksecattraccessgroup]</sup> for more details). Access to the Keychain is managed by the `securityd` daemon, which grants access based on the app's `Keychain-access-groups`, `application-identifier` and `application-group` entitlements (More information can be found within the Apple documentation <sup>[https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppDistributionGuide/AddingCapabilities/AddingCapabilities.html, https://developer.apple.com/library/content/documentation/Security/Conceptual/keychainServConcepts/02concepts/concepts.html]</sup>).
```
Comment (commjoen) [inline Document/0x06d-Testing-Data-Storage.md]: Fixed!

File: Document/0x06d-Testing-Data-Storage.md (hunk 1 above)
Comment (commjoen) [inline Document/0x06d-Testing-Data-Storage.md]: Fixed

**Pull request merged**